src/psse_export.jl

```diff
@@ -0,0 +1,599 @@
+    # We could also strip trailing empty (default) entries here
+    join(io, iterator, delim)
+    println(io)
+end
+
+_permissive_parse_int(x) = Int64(parse(Float64, x))  # Parses "1.0" as 1, errors on "1.5"
+
+_psse_quote_string(s::String) = "'$s'"
+
+"Throw a `NotImplementedError` if the `psse_version` is not `:v33`"
+check_33(psse_version::Symbol) =
+    (psse_version == :v33) ||
+    throw(IS.NotImplementedError("Only implemented for psse_version $(:v33)"))
+
+function _validate_container_number(unparsed::String)
+    parsed = try
+        _permissive_parse_int(unparsed)
+    catch e
+        if e isa Union{InexactError, ArgumentError}
+            throw(ArgumentError("container name $unparsed could not be parsed as an integer"))
+        else
+            rethrow(e)
+        end
+    end
+    (parsed in 1:9999) && (return parsed)
+    throw(ArgumentError("container number $parsed is out of range"))
+end
+
+"Validate that the Sienna area/zone names parse as PSS/E-compatible area/zone numbers, output a mapping"
+_psse_container_numbers(container_names::Vector{String}) =
+    DataStructures.OrderedDict(
+        name => _validate_container_number(name) for name in container_names
+    )
+
+"WRITTEN TO SPEC: PSS/E 33.3 POM 5.2.1 Case Identification Data"
+function _write_case_identification_data(
+    io::IO,
+    md::AbstractDict,
+    sys::System,
+    psse_version::Symbol,
+    case_name::String,
+)
+    check_33(psse_version)
+
+    # Record 1
+    IC = 0
+    SBASE = PSY.get_base_power(sys)
+    REV = 33
+    XFRRAT = 0
+    NXFRAT = 1  # TODO why?
+    BASFRQ = PSY.get_frequency(sys)
+    joinln(io, [IC, SBASE, REV, XFRRAT, NXFRAT, BASFRQ])
+
+    # Record 2
+    (length(case_name) <= 60) ||
+        throw(ArgumentError("case_name may be up to 60 characters"))
+    println(io, case_name)
+
+    # Record 3
+    line3 = "Created by PowerFlows.jl on $(Dates.now())"
+    @assert length(line3) <= 60
+    println(io, line3)
+end
+
+"""
+Given a vector of Sienna bus numbers, create a dictionary from Sienna bus number to
+PSS/E-compatible bus number. Assumes that the Sienna bus numbers are positive and unique.
+Guarantees determinism: if the input contains the same bus numbers in the same order, the
+output will. Guarantees minimal changes: that if an existing bus number is compliant, it
+will not be changed.
+
+WRITTEN TO SPEC: PSS/E 33.3 POM 5.2.1 Bus Data
+"""
+function _psse_bus_numbers(buses::Vector{Int64})
+    used_numbers = Set{Int64}()
+    sizehint!(used_numbers, length(buses))
+    mapping = OrderedDict{Int64, Int64}()
+    sizehint!(mapping, length(buses))
+
+    for original_number in buses
+        if original_number in 1:999_997
+            mapping[original_number] = original_number
+            push!(used_numbers, original_number)
+        end
+    end
+    for original_number in buses
+        haskey(mapping, original_number) && continue
+        new_number = original_number
+        new_number %= 1_000_000
```

Comment:
we should consider addressing this in the parser too. I understand the safety logic of never creating a bus number that PSSe can recognize but it makes a mess to write back